cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -180,11 +190,22 @@ func (c *Client) GetMemberID(peerURL string) (uint64, error) {
 	}
 	defer cli.Close()
 
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	resp, err := cli.MemberList(ctx)
-	cancel()
+	// Gets the member list
+	var lastError error
+	var resp *clientv3.MemberListResponse
+	err = wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
The 5 second parameter should probably be a const somewhere.

Reply:
the 5 seconds here are not needed, you can just do `cli.Member<METHOD>(context.Background())` for all member functions.

as long as the retry timeout is significant we do not care about the per call timeout.

Reply:
as per slack discussion, timeouts are a safeguard for avoiding kubeadm be stuck waiting for etcd.